srml/elections-phragmen/src/lib.rs

```diff
@@ -0,0 +1,957 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! An election module based on sequential phragmen.
+//!
+//! The election happens in _rounds_: every `N` blocks, the all previous members are retired and a
+//! new set is elected (which may or may not have some of the same members). Each round lasts for
+//! some number of blocks defined by `TermDuration` storage item. Each cycle of election is a term.
+//!
+//! `TermDuration` might change during a round. This can shorten or extend the length of the round.
+//! The next election round's block number is never stored but rather always checked on the fly.
+//! Based on the current block number and `TermDuration`, `BlockNumber % TermDuration == 0` will
+//! always trigger a new election round.
+//!
+//! Voters can vote for as many of the candidates by providing a list of account ids. Invalid votes
+//! (voting for non-candidates) are ignored during election. Voters reserve a bond as they vote.
+//! The entire free balance of a voter is locked upon voting and can only be used to pay for fees.
+//! Voters can update their votes by calling `vote()` again during the same round. This does not
+//! effect the reserved bond or lock. After a term, voters _must_ call `remove_voter` to get their
+//! bond back and remove the lock. Otherwise the bond amount will stay reserved and the lock will
+//! persist. Furthermore, voters of an old round cannot vote for the current round until they remove
+//! their previous data via `remove_voter`.
+//!
+//! Candidates also reserve a bond as they submit candidacy. A candidate cannot take their candidacy
+//! back. Winner candidates will be moved to the members list and will get their bond back at end of
+//! their term. Loser candidates will immediately get their bond back. Note that unlike phragmen in
+//! staking, candidates do NOT automatically vote for themselves (though they _could_ via a separate
+//! transaction). Furthermore, the amount of tokens (stake) held by the candidate does not matter.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use sr_primitives::traits::{Zero, StaticLookup, Bounded, Convert};
+use sr_primitives::weights::SimpleDispatchInfo;
+use srml_support::{StorageValue, StorageMap, EnumerableStorageMap, decl_storage, decl_event, ensure,
+	decl_module, dispatch,
+	traits::{Currency, Get, LockableCurrency, LockIdentifier, ReservableCurrency, WithdrawReason,
+		WithdrawReasons, ChangeMembers
+	}
+};
+use system::{self, ensure_signed, ensure_root};
+
+const MODULE_ID: LockIdentifier = *b"phrelect";
+
+type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
+
+pub const DEFAULT_CANDIDACY_BOND: u32 = 9;
```

Comment:
wtf are these? units? docs?

Reply:
idk these were added as configs to council modules some few weeks ago e.g.

https://github.com/paritytech/substrate/blob/a622361559b31f1eb9eb8ce222ad6f94bf4b79c4/srml/elections/src/lib.rs#L144-L151

Reply:
but now I see they are unused in both places apparently. Will clean up.